src/Gallery.js

```diff
@@ -0,0 +1,99 @@
+import React, { Component } from 'react'
+import { STATUS } from './constants';
+import { setTimeout } from 'timers';
+
+class Gallery extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            cats: [],
+            loadingState: STATUS.FETCHING,
+            autoScroll: false,
+        }
+
+        this.setScroll = this.setScroll.bind(this);
+        this.handleScroll = this.handleScroll.bind(this);
+        this.scrollToBottom = this.scrollToBottom.bind(this);
+        this.handlecheckbox = this.handlecheckbox.bind(this);
+        this.fetchCatsCollection = this.fetchCatsCollection.bind(this);
+    };
+
+    componentDidMount() {
+        this.fetchCatsCollection();
+        window.addEventListener('scroll', this.handleScroll)
+    }
+
+    componentWillUnmount(){
+        
+    }
+
+    scrollToBottom() {
+        this.pageEnd.scrollIntoView({ behavior: "smooth" });
+    }
+
+    handleScroll(e) {
+        if (Math.floor(e.target.scrollingElement.scrollHeight - e.target.scrollingElement.scrollTop) === e.target.scrollingElement.clientHeight) {
+            this.fetchCatsCollection()
+        }
+    }
+
+    fetchCatsCollection() {
+        this.setState({ loadingState: STATUS.FETCHING })
+        fetch("https://api.thecatapi.com/v1/images/search?limit=10", {
+            headers: {
+                "Content-Type": "application/json",
+                "x-api-key": "4bebae0d-0ec4-4787-8e77-8602741525af"
+            }
+        })
+            .then(data => data.json())
+            .then(data => {
+                data.map(item => {
+                    const { url } = item;
+                    this.setState({ cats: this.state.cats.concat(url), loadingState: STATUS.FETCHED });
+                })
+            });
+    }
+
+    handlecheckbox(e) {
+        this.setState({ autoScroll: e.target.checked }, () => {
+            this.setScroll()
+        })
+    }
+
+    setScroll() {
+        if (this.state.autoScroll) {
+            setTimeout(() => {
+                this.pageEnd.scrollIntoView({ behavior: "smooth" });
+            }, 10 * 1000)
+        }
+    }
+
+    render() {
+        return (
+            <React.Fragment>
+                <div>
+                    Autoscroll <input type='checkbox' onChange={this.handlecheckbox} />
+                    <div className='photos'>
+                        {this.state.cats.map(cat => {
+                            return < img
+                                style={{
```

Comment:
The problem with blinking after scrolling to the bottom of the page is caused by this line - it was ok to use it for Random Cat route but in this one it hides all the images when the status changes.

Once this line is removed everything works as expected 👍